core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/OnDemandRepairJob.java

```diff
@@ -156,9 +170,11 @@ public void postExecute(boolean successful, ScheduledTask task)
             LOG.error("Error running {}", task);
             failed = true;
         }
-
-        Set<LongTokenRange> repairedTokenSet = myTasks.remove(task);
```

Comment:
Wouldn't this never removed failed jobs?

Reply:
No, this only removes the task for that token range and since the job failed it will not be retried anyway.